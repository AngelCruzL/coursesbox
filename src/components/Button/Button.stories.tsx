import type { Meta, StoryObj } from "@storybook/react";

import { Button } from "./Button";

const meta: Meta<typeof Button> = {
  title: "Controls/Button",
  component: Button,
  tags: ["autodocs"],
  argTypes: {},
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: "Primary Button",
  },
};

export const Secondary: Story = {
  args: {
    children: "Secondary Button",
    color: "secondary",
  },
};

export const Danger: Story = {
  args: {
    children: "Danger Button",
    color: "danger",
  },
};

export const Warning: Story = {
  args: {
    children: "Warning Button",
    color: "warning",
  },
};
