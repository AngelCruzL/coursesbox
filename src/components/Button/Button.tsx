import styled from "@emotion/styled";

export const Button = styled.button`
  all: unset;
  display: flex;
  justify-self: center;
  align-items: center;
  justify-content: center;
  user-select: none;
  cursor: pointer;
  font-size: 1.6rem;
  inline-size: 15rem;
  block-size: 4rem;
  border-radius: 1rem;
  transition: all 0.4s ease;

  &:hover {
    opacity: 0.9;
  }

  box-shadow: 0.5vmin 0.5vmin 1vmin #c8d0e7, -0.5vmin -0.5vmin 1vmin #fff;
`;
