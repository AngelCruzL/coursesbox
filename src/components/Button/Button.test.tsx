import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Button } from "./Button";

describe("Button test cases", () => {
  let onClick: () => void;

  beforeEach(() => {
    onClick = jest.fn();
  });

  it("should render successfully", () => {
    const { asFragment } = render(<Button onClick={onClick}>Button</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should check if button is clicked", async () => {
    render(<Button onClick={onClick}>Button</Button>);
    const button = screen.getByRole("button");
    await userEvent.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  test("should render primary button", () => {
    const { asFragment } = render(
      <Button color="primary" onClick={onClick}>
        Primary Button
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render secondary button", () => {
    const { asFragment } = render(
      <Button color="secondary" onClick={onClick}>
        Secondary Button
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render danger button", () => {
    const { asFragment } = render(
      <Button color="danger" onClick={onClick}>
        Danger Button
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render warning button", () => {
    const { asFragment } = render(
      <Button color="warning" onClick={onClick}>
        Warning Button
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
