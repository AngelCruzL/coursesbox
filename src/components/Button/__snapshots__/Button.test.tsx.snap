// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases should render danger button 1`] = `
<DocumentFragment>
  <button
    class="css-18eoxov"
    color="danger"
  >
    Danger Button
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render primary button 1`] = `
<DocumentFragment>
  <button
    class="css-1qkvrts"
    color="primary"
  >
    Primary Button
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render secondary button 1`] = `
<DocumentFragment>
  <button
    class="css-1tpzt5k"
    color="secondary"
  >
    Secondary Button
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render successfully 1`] = `
<DocumentFragment>
  <button
    class="css-1qkvrts"
    color="primary"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`Button test cases should render warning button 1`] = `
<DocumentFragment>
  <button
    class="css-b3yccl"
    color="warning"
  >
    Warning Button
  </button>
</DocumentFragment>
`;
